File: Selenium/Project/ProjectActivity9.java
package Selenium_Java_Project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;

import java.time.Duration;

public class ProjectActivity9 {

    public static void main(String args[]) {

        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://alchemy.hguy.co/jobs/wp-admin");

        driver.findElement(By.id("user_login")).sendKeys("root");
        driver.findElement(By.id("user_pass")).sendKeys("pa$$w0rd");
        driver.findElement(By.id("wp-submit")).click();

        driver.findElement(By.xpath("//div[text()='Job Listings ']")).click();

        driver.findElement(By.xpath("//a[contains(@class, 'page-title-action')]")).click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//button[text()='Next']")));

        driver.findElement(By.xpath("//button[text()='Next']")).click();
        driver.findElement(By.xpath("//button[text()='Next']")).click();
        driver.findElement(By.xpath("//button[text()='Next']")).click();

        driver.findElement(By.xpath("//button[contains(@class, 'components-button components-guide__finish-button is-primary')]")).click();

        driver.findElement(By.xpath("//textarea[@id='post-title-0']")).sendKeys("Test specialist");

        driver.findElement(By.xpath("//div[contains(@class, 'block-library-classic__toolbar has-advanced-toolbar')]")).click();

        driver.findElement(By.xpath("//div[starts-with(@id, 'editor-')]")).sendKeys("Automating the app");

        driver.findElement(By.xpath("//button[text()='Publish…']")).click();
        driver.findElement(By.xpath("//button[text()='Publish']")).click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(@class, 'components-button is-secondary')]")));
        driver.findElement(By.xpath("//a[contains(@class, 'components-button is-secondary')]")).click();
        String jobTitleText = driver.findElement(By.xpath("//h1[text()='Test specialist']")).getText();

        Assert.assertEquals(jobTitleText, "Test specialist");

        driver.quit();







    }
}

File: Python/Activities/ProjectActivity9.py
# Set up the Firefox Driver with WebDriverManger
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

service1 = FirefoxService(GeckoDriverManager().install())


# Start the Driver
with webdriver.Firefox(service=service1) as driver:

    driver.get("https://alchemy.hguy.co/jobs/wp-admin")

    driver.find_element(By.XPATH, "//input[@id='user_login']").send_keys("root")

    driver.find_element(By.XPATH, "//input[@id='user_pass']").send_keys("pa$$w0rd")

    driver.find_element(By.XPATH, "//input[@id='wp-submit']").click()

    pageTitle = driver.title

    print("Title of the page: ", pageTitle)

    assert pageTitle == "Dashboard ‹ Alchemy Jobs — WordPress"

    driver.find_element(By.XPATH, "//div[text()='Job Listings ']").click()

    driver.find_element(By.XPATH, "//a[contains(@class, 'page-title-action')]").click()

    driver.find_element(By.XPATH, "//button[text()='Next']").click()
    driver.find_element(By.XPATH, "//button[text()='Next']").click()
    driver.find_element(By.XPATH, "//button[text()='Next']").click()

    driver.find_element(By.XPATH, "//button[contains(@class, 'components-button components-guide__finish-button is-primary')]").click()

    driver.find_element(By.XPATH, "//textarea[@id='post-title-0']").send_keys("Testor")

    wait = WebDriverWait(driver, 10)



    driver.find_element(By.XPATH, "//div[@id='toolbar-1a0567ea-1c28-4311-8a59-009e7b9b62ac']").click()

    driver.find_element(By.XPATH, "//div[starts-with(@id, 'editor-')]").send_keys("QA")

    driver.find_element(By.XPATH, "//input[@id='_job_location']").send_keys("Flipkart")

    driver.find_element(By.XPATH, "//button[text()='Publish…']").click()

    driver.find_element(By.XPATH, "//button[text()='Publish']").click()

    driver.find_element(By.XPATH, "//a[contains(@class, 'components-button is-secondary')]").click()

    Title = driver.find_element(By.XPATH, "//h1[contains(@class, 'entry-title')]")

    print("Entry title: ", Title.text)

    entryTitle = Title.text

    assert entryTitle == "Testor"

